subsys/bluetooth/host/conn.c

```diff
@@ -3657,4 +3678,14 @@ void bt_hci_le_df_cte_req_failed(struct net_buf *buf)
 }
 #endif /* CONFIG_BT_DF_CONNECTION_CTE_REQ */
 
+#else
+
+int bt_conn_disconnect(struct bt_conn *conn, uint8_t reason)
+{
+	ARG_UNUSED(conn);
+	ARG_UNUSED(reason);
+
+	return 0;
+}
```

Comment:
Why do we (sort of) implement this function when `CONFIG_BT_CONN=n`?

Reply:
talk to the compiler/linker and build system I guess.

Reply:
Wut :D 

Assuming that all calls to this is guarded by `CONFIG_BT_CONN` (with #if or IS_ENABLED) we shouldn't have any issues.

Reply:
I removed it. But had to add something dumb: https://github.com/jori-nordic/zephyr/blob/buffer-rewrite-patchset/subsys/bluetooth/host/conn.c#L945-L949

Reply:
This may also be a side effect of the... "improvement"... I did a while ago for broadcast ISO. 

ISO uses a lot of things from conn.c, and thus when you are a ISO broadcast source only, you still use `conn.c` while `CONFIG_BT_CONN=n`. 

This could be the reason why some things are as expected, but also why the above "fix" you've added won't work: 
ISO Broadcasters will (likely?) use `bt_conn_tx_processor`, but `CONFIG_BT_CONN` will be `n`, and thus the above may not work.

Reply:
ah shoot, that sucks. So I just have to litter the code with `IS_ENABLED(BT_CONN)`?
I guess we'll see if the CI fails, we have tests for ISO broadcasters-only, riiiiight?

(I think so, the tests that were failing with linker errors were the `audio_samples/` and they seem to be building and running broadcaster-only devices, unless I'm mistaken)

Reply:
> I guess we'll see if the CI fails, we have tests for ISO broadcasters-only, riiiiight?

We have the tests for the broadcast audio samples that should fail, assuming that they are run. 

Otherwise we don't actually have any BIS-only CI tests except the one I'm adding in https://github.com/zephyrproject-rtos/zephyr/pull/72690